test/blackbox/common/TCPReqRepHelloWorldReplier.hpp

```diff
@@ -118,14 +119,53 @@ class TCPReqRepHelloWorldReplier
     }
     reply_listener_;
 
+    class RequestBussyListener : public eprosima::fastrtps::PublisherListener
```

Comment:
Instead of adding a new class, I would make the current `RequestListener` call a new `unmatched()` method in the replier, that will sleep when `use_busy_listener` is `true`